diff -Naru libpng-1.6.39/png.c libpng-1.6.39.aros/png.c
--- libpng-1.6.39/png.c	2022-11-20 22:53:23.000000000 +0100
+++ libpng-1.6.39.aros/png.c	2024-09-29 22:16:27.345345621 +0200
@@ -700,7 +700,7 @@
  * function of your own because "FILE *" isn't necessarily available.
  */
 void PNGAPI
-png_init_io(png_structrp png_ptr, FILE *fp)
+__png_init_io(png_structp png_ptr, FILE *fp, png_rw_ptr read_data_fn, png_rw_ptr write_data_fn, png_flush_ptr output_flush_fn)
 {
    png_debug(1, "in png_init_io");
 
@@ -708,6 +708,10 @@
       return;
 
    png_ptr->io_ptr = (png_voidp)fp;
+   if (png_ptr->mode & PNG_IS_READ_STRUCT)
+      png_set_read_fn(png_ptr, fp, read_data_fn);
+   else
+      png_set_write_fn(png_ptr, fp, write_data_fn, output_flush_fn);
 }
 #  endif
 
